File: src/dotenv_ffi.erl
-module(dotenv_ffi).
-export([get_env/1, set_env/2]).

get_env(Key) ->
    case os:getenv(binary_to_list(Key)) of
        false -> {error, nil};
        Value -> {ok, list_to_binary(Value)}
    end.

set_env(Key, Value) ->
    os:putenv(binary_to_list(Key), binary_to_list(Value)),
    true.

File: src/dotenv_ffi.mjs
export function getEnv(key) {
  const value = process.env[key];
  return value !== undefined ? { tag: "Ok", value } : { tag: "Error", error: null };
}

export function setEnv(key, value) {
  process.env[key] = value;
  return true;
}
